core/src/main/java/com/sequenceiq/cloudbreak/service/AbstractWorkspaceAwareResourceService.java

```diff
@@ -52,18 +52,22 @@ public T create(T resource, Workspace workspace, User user) {
             prepareCreation(resource);
             setWorkspace(resource, user, workspace);
             return repository().save(resource);
-        } catch (DataIntegrityViolationException e) {
-            String message = String.format("%s already exists with name '%s' in workspace %s",
-                    resource().getShortName(), resource.getName(), resource.getWorkspace().getName());
-            throw new BadRequestException(message, e);
+        } catch (AccessDeniedException e) {
+            ConstraintViolationException cve = ThrowableUtil.getSpecificCauseRecursively(e, ConstraintViolationException.class);
+            if (cve != null) {
+                String message = String.format("%s already exists with name '%s' in workspace %s",
+                        resource().getShortName(), resource.getName(), resource.getWorkspace().getName());
+                throw new AccessDeniedException(message, e);
+            }
+            throw e;
         }
     }
 
     @Override
     public T getByNameForWorkspace(String name, Workspace workspace) {
         T object = repository().findByNameAndWorkspace(name, workspace);
         if (object == null) {
-            throw new NotFoundException(String.format("No %s found with name '%s'", resource().getShortName(), name));
+            throw new AccessDeniedException(String.format("No %s found with name '%s'", resource().getShortName(), name));
```

Comment:
Both NotFound and AccessDenied exceptions are mapped to 403 Forbidden, so it shouldn't matter which one we throw.